tuxemon/player.py

```diff
@@ -50,6 +50,13 @@ def update(self, time_delta: float) -> None:
         diff_y = abs(after.y - before.y)
 
         self.game_variables["steps"] += diff_x + diff_y
+
+        for key, value in self.game_variables.items():
+            if key.startswith("steps_") and float(value) > 0.0:

```

Comment:
Since the first predicate of the condition is in both cases the same, and the second predicate is just reversed in the other case, we could simplify the whole block by this:

```py
if key.startswith("steps_"):
    if float(value) > 0.0:
        self.game_variables[key] = float(value) - (diff_x + diff_y)
    else:
        self.game_variables[key] = str(0)
```

Also since in both cases it's assigning a value to the same identifier, we could choose to use "ternary" operator, but I let it up to you.

Reply:
Btw, why `str(0)`? Why not just "0" or even 0.0, since in the other case it's a floating value that is assigned.

Reply:
@Grimmys your way is better

regarding str(0), that's because of `variable_set` is set default str()